'No such dataset' in klaas_split_data with current master

Hi, 
when running klaas_split_data from the current master, I get the following error message:
 
```
klaas_split_data build/data/gamma_precuts.hdf5 build/data/gamma \
        -f 0.25 -n train_sep \
        -f 0.5 -n train_reg \
        -f 0.25 -n test \
        -i events
INFO:root:Found a total of 692836 single-telescope events in the file
INFO:root:Writing 173209 telescope-array events to: build/data/gamma_train_sep.hdf5
Traceback (most recent call last):
  File "/home/jbuss/.conda/envs/gamma/bin/klaas_split_data", line 11, in <module>
    load_entry_point('klaas', 'console_scripts', 'klaas_split_data')()
  File "/home/jbuss/.conda/envs/gamma/lib/python3.5/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/home/jbuss/.conda/envs/gamma/lib/python3.5/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/home/jbuss/.conda/envs/gamma/lib/python3.5/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/jbuss/.conda/envs/gamma/lib/python3.5/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/home/jbuss/software/classifier-tools/klaas/scripts/split_data.py", line 111, in main
    write_data(selected_data, path, key=key, use_hp5y=not use_pandas)
  File "/home/jbuss/software/pyfact/fact/io.py", line 34, in write_data
    to_h5py(file_path, df, key=key, **kwargs)
  File "/home/jbuss/software/pyfact/fact/io.py", line 267, in to_h5py
    append_to_h5py(f, array, key=key)
  File "/home/jbuss/software/pyfact/fact/io.py", line 428, in append_to_h5py
    raise KeyError('No such dataset {}'.format(dataset))
KeyError: 'No such dataset None'
make: *** [build/data/gamma_train_reg.hdf5] Fehler 1
```

hmmm, but also an error if I use the pandas option

```
Traceback (most recent call last):
  File "/home/jbuss/.conda/envs/gamma/lib/python3.5/site-packages/pandas/core/indexing.py", lin
e 1794, in _get_list_axis
    return self.obj._take(key, axis=axis, convert=False)
  File "/home/jbuss/.conda/envs/gamma/lib/python3.5/site-packages/pandas/core/generic.py", line
 2150, in _take
    verify=True)
  File "/home/jbuss/.conda/envs/gamma/lib/python3.5/site-packages/pandas/core/internals.py", li
ne 4250, in take
    indexer = maybe_convert_indices(indexer, n)
  File "/home/jbuss/.conda/envs/gamma/lib/python3.5/site-packages/pandas/core/indexing.py", lin
e 2103, in maybe_convert_indices
    raise IndexError("indices are out-of-bounds")
IndexError: indices are out-of-bounds

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/jbuss/.conda/envs/gamma/bin/klaas_split_data", line 11, in <module>
    load_entry_point('klaas', 'console_scripts', 'klaas_split_data')()
  File "/home/jbuss/.conda/envs/gamma/lib/python3.5/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/home/jbuss/.conda/envs/gamma/lib/python3.5/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/home/jbuss/.conda/envs/gamma/lib/python3.5/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/jbuss/.conda/envs/gamma/lib/python3.5/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/home/jbuss/software/classifier-tools/klaas/scripts/split_data.py", line 118, in main
    data = data.iloc[list(set(data.index.values) - set(selected_data.index))]
  File "/home/jbuss/.conda/envs/gamma/lib/python3.5/site-packages/pandas/core/indexing.py", line 1373, in __getitem__
    return self._getitem_axis(maybe_callable, axis=axis)
  File "/home/jbuss/.conda/envs/gamma/lib/python3.5/site-packages/pandas/core/indexing.py", line 1819, in _getitem_axis
    return self._get_list_axis(key, axis=axis)
  File "/home/jbuss/.conda/envs/gamma/lib/python3.5/site-packages/pandas/core/indexing.py", line 1797, in _get_list_axis
    raise IndexError("positional indexers are out-of-bounds")
IndexError: positional indexers are out-of-bounds
make: *** [build/data/gamma_train_reg.hdf5] Fehler 1
```

--- a/klaas/scripts/split_data.py
+++ b/klaas/scripts/split_data.py
@@ -115,7 +115,7 @@ def main(input_path, output_basename, fraction, name, inkey, key, event_id_key, 
             log.info('Writing {} telescope-array events to: {}'.format(n, filename))
             selected_data.to_csv(filename, index=False)
 
-        data = data.iloc[list(set(data.index.values) - set(selected_data.index))]
+        data = data.loc[list(set(data.index.values) - set(selected_data.index.values))]
 
         if event_id_key:
             ids = data[event_id_key].unique()

--- a/setup.py
+++ b/setup.py
@@ -2,7 +2,7 @@ from setuptools import setup
 
 setup(
     name='klaas',
-    version='0.7.1',
+    version='0.7.3',
     description='KLAssification And regression Scripts.  yay',
     url='https://github.com/fact-project/klaas',
     author='Kai Brügge, Maximilian Nöthe, Jens Buss',

